client/my-sites/checkout/composite-checkout/lib/update-cart-contact-details-for-checkout.ts

```diff
@@ -43,5 +43,7 @@ export async function updateCartContactDetailsForCheckout(
 		vatId: vatDetails.id ?? '',
 		organization: vatDetails.name ?? '',
 		address: vatDetails.address ?? '',
+		city: contactInfo.city?.value ?? '',
+		state: contactInfo.state?.value ?? '',
```

Comment:
Something we need to solve, although I'm not sure of the exact place to solve it: even if the fields are hidden, the data for these fields remains in the store. If it was previously filled (eg: via auto-fill from a previous entry or manually before the country was changed) then the (now incorrect) fields will still be sent to the shopping cart.

For example, here's what happens if you fill in the form for Canada, but then change it to Jamaica. The shopping-cart endpoint receives this payload:

<img width="211" alt="Screenshot 2023-02-09 at 3 53 11 PM" src="https://user-images.githubusercontent.com/2036909/217937568-5b57af1f-b79e-4588-a8a7-a54924bc8fb8.png">

Reply:
Good point. I should be able to fix this and make `onChange` more typesafe at the same time.